File: scala/src/test/scala/com/haiboyan/scala/algorithm/array/ZigzagTest.scala
package com.haiboyan.scala.algorithm.array

import org.scalatest.{FunSuite, Matchers}

class ZigzagTest extends FunSuite with Matchers{
  test("Zigzag abc on 2 should be acb") {
    assert(Zigzag.convert("abc", 2) === "acb")
  }

  test("Zigzag abc on 3 should be acb") {
    assert(Zigzag.convert("abc", 3) === "abc")
  }
}

File: go/tests/array/zigzag_test.go
package array

import (
	"github.com/go/array"
	"testing"
)

func TestZigzag(t *testing.T) {
	if array.Convert("abc", 2) != "acb" {
		t.Errorf("Zigzag abc, 2 should rerturn acb.")
	}

	if array.Convert("abc", 3) != "abc" {
		t.Errorf("Zigzag abc, 3 should rerturn abc.")
	}

	if array.Convert("PAYPALISHIRING", 3) != "PAHNAPLSIIGYIR" {
		t.Errorf("Zigzag PAYPALISHIRING, 3 should rerturn PAHNAPLSIIGYIR.")
	}
}
